src/snowflake/snowpark/mock/_stored_procedure.py

```diff
@@ -0,0 +1,176 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+import sys
+import typing
+from types import ModuleType
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+import snowflake.snowpark
+from snowflake.snowpark._internal.analyzer.expression import Literal
+from snowflake.snowpark._internal.udf_utils import (
+    check_python_runtime_version,
+    process_registration_inputs,
+)
+from snowflake.snowpark._internal.utils import TempObjectType
+from snowflake.snowpark.column import Column
+from snowflake.snowpark.exceptions import SnowparkSQLException
+from snowflake.snowpark.stored_procedure import (
+    StoredProcedure,
+    StoredProcedureRegistration,
+)
+from snowflake.snowpark.types import DataType, _FractionalType, _IntegralType
+
+from ._telemetry import LocalTestOOBTelemetryService
+
+if sys.version_info <= (3, 9):
+    from typing import Iterable
+else:
+    from collections.abc import Iterable
+
+
+def sproc_types_are_compatible(x, y):
+    if isinstance(x, type(y)):
+        return True
+    elif isinstance(x, _IntegralType) and isinstance(y, _IntegralType):
+        return True
+    elif isinstance(x, _FractionalType) and isinstance(y, _FractionalType):
+        return True
+    return False
+
+
+class MockStoredProcedure(StoredProcedure):
+    def __call__(
+        self,
+        *args: Any,
+        session: Optional["snowflake.snowpark.session.Session"] = None,
+        statement_params: Optional[Dict[str, str]] = None,
+    ) -> Any:
+        args, session = self._validate_call(args, session)
+
+        # Unpack columns if passed
+        parsed_args = []
+        for arg, expected_type in zip(args, self._input_types):
+            if isinstance(arg, Column):
```

Comment:
Question: why do we need this case? Do we support column as a stored procedure argument?